hpx/util/unwrap.hpp

```diff
@@ -0,0 +1,166 @@
+//  Copyright (c) 2017 Denis Blank
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_UTIL_UNWRAP_HPP
+#define HPX_UTIL_UNWRAP_HPP
+
+#include <hpx/config.hpp>
+#include <hpx/util/detail/unwrap_impl.hpp>
+
+#include <cstddef>
+#include <utility>
+
+namespace hpx {
+namespace util {
+    /// A helper function for retrieving the actual result of
+    /// any hpx::lcos::future like type which is wrapped in an arbitrary way.
+    ///
+    /// Unwraps the given pack of arguments, so that any hpx::lcos::future
+    /// object is replaced by its future result type in the argument pack:
+    /// - `hpx::future<int>` -> `int`
+    /// - `hpx::future<std::vector<float>>` -> `std::vector<float>`
+    /// - `std::vector<future<float>>` -> `std::vector<float>`
+    ///
```

Comment:
Yes, it would. Also futures nested inside arbitrary combinations of tuples, pairs and homogeneous containers like `std::list` or `std::vector` are being remapped. For instance;

```cpp
std::list<
  std::pair<
    int,
    std::tuple<
      int,
      std::vector<
        hpx::lcos::future<int>
      >,
      double
    >
  >
>
```

is also remapped.